zebra-chain/src/orchard/tests/prop.rs

```diff
@@ -0,0 +1,43 @@
+use proptest::prelude::*;
+use std::io::Cursor;
+
+use super::super::super::*;
+
+use crate::serialization::{ZcashDeserializeInto, ZcashSerialize};
+
+proptest! {
+    #[test]
+    // Make sure only valid flags deserialize
+    fn flags_roundtrip(flags in any::<u8>()) {
+
+        let mut serialized = Cursor::new(Vec::new());
+        flags.zcash_serialize(&mut serialized)?;
+
+        serialized.set_position(0);
+        let maybe_deserialized = (&mut serialized).zcash_deserialize_into();
+
+        let mut error_catch = String::new();
+        let maybe_deserialized: Option<orchard::Flags> = match maybe_deserialized {
+            Ok(deserialized) => Some(deserialized),
+            Err(e) => {
+                error_catch = e.to_string();
+                None
+            },
+        };
+
+        let binary_repr = format!("{:b}", flags);
+        if maybe_deserialized.is_some() {
+            let flag = maybe_deserialized.unwrap();
+            prop_assert!(binary_repr.len() <= 2);
+            prop_assert!(
+                flag == orchard::Flags::empty()
+                || flag == orchard::Flags::ENABLE_SPENDS
+                || flag == orchard::Flags::ENABLE_OUTPUTS
+                || flag == orchard::Flags::ENABLE_SPENDS | orchard::Flags::ENABLE_OUTPUTS);
+        }
+        else {
+            prop_assert!(binary_repr.len() > 2);
+            prop_assert_eq!(error_catch, "parse error: invalid orchard flags");
+        }
+    }
```

Comment:
I'm not sure if this is a good suggestion, but maybe it could help with ideas to make it simpler:
```suggestion
        let invalid_bits_mask = !orchard::Flags::all().bits();
        match orchard::Flags::from_bits(flags) {
            Some(valid_flags) => {
                prop_assert_eq!(maybe_deserialized, Ok(valid_flags));
                prop_assert_eq!(flags & invalid_bits_mask, 0);
            }
            None => {
                prop_assert_eq!(maybe_deserialized, Err(SerializationError::Parse("invalid orchard flags")));
                prop_assert_ne!(flags & invalid_bits_mask, 0);
            }
        }
    }
```

Reply:
I like it, thanks :)